backend/src/repositories/booking.repository.ts

```diff
@@ -0,0 +1,9 @@
+import { Booking, BookingStatus } from "../models/booking.model";
+
+interface IBookingRepository {
+  save(booking: Booking) : Promise<Booking>;
+  update(booking: Booking) : Promise<Booking>;
+  findById(bookingId : number) : Promise<Booking>;
+  findByUser(userId: number) : Promise<Booking[]>;
+  findAll(searchCriteria: {userId?: number, bikeId?: number, status?: BookingStatus}) : Promise<Booking[]>
```

Comment:
For now, I think it will be more appropriate to use temporarily userName instead of userId. As we will not have a login at the beginning, searching for userId probably will be ineffective.